tests/functional/conftest.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+import pytest
+import webtest
+
+from tests.simple_server import serve_content
+from viahtml.app import Application
+
+
+@pytest.fixture(scope="session")  # pragma: no cover
```

Comment:
I've no idea what's up with the coverage, but once you've created the main app all hell breaks lose and nothing makes sense any more. These fixture lines were apparently not covered, but that's total nonsense as they are declarative.

Reply:
I'm wondering about the session scope of this and the `with_environ` fixture below. Could this couple tests together in such a way that later tests depend on earlier ones having been run before? In theory both fixtures definitely could do

Reply:
I don't think so. As this returns an app the tests are making requests to it which are individual. Our app shouldn't store state between requests that could modify the next.

We actually have a single baked request for most of the tests, so this further reduces the chance of any state between requests, as there is only one for the majority. As long as we don't mutate the response in our tests they can't become dependent.